### PR TITLE
get all implementors for a class

### PR DESCRIPTION
**Pull request by keertip**
@sethladd

### CODE REVIEW

File: lib/src/model.dart [hunk 1]
```diff
@@ -8,44 +8,55 @@ library dartdoc.models;
...
+Map<Class, List<Class>> _implementors = new Map();
+
 /// Returns all the implementors of the class specified.
-// TODO: should this go into model_utils.dart ?
-// TODO(keertip): impelement this
-List<Class> getAllImplementorsFor(Class c) =>
-    [new MockClass('SampleClass'), new MockClass('AnotherClass')];
+List<Class> getAllImplementorsFor(Class c) {
+  var m = _implementors[c];
+  return m;}
```
Comment (sethladd) [inline lib/src/model.dart]: put } on new line?

File: lib/src/model.dart [hunk 2]
```diff
@@ -8,44 +8,55 @@ library dartdoc.models;
...
+
+void _addToImplementors(Class c) {
+  void _checkAndAddClass(Class key, Class implClass) {
+    List list = _implementors[key];
+    if (list == null) {
+      list = new List();
```
Comment (sethladd) [inline lib/src/model.dart]: can't wait for the elvis operator!! :)

File: lib/src/model.dart [hunk 3]
```diff
@@ -469,20 +483,24 @@ class Library extends ModelElement {
...
+        .toList(growable: true);
 
     return _classes;
   }
 
-  // TODO: rename this to getClasses
-  List<Class> getTypes() {
-    return allClasses.where((c) => !c.isErrorOrException).toList(growable:false);
+  List<Class> getClasses() {
+    return allClasses
```
Comment (sethladd) [inline lib/src/model.dart]: did you update the template, too?

File: lib/src/model.dart (hunk 2 above)
Comment (keertip) [inline lib/src/model.dart]: :)

File: lib/src/model.dart (hunk 3 above)
Comment (keertip) [inline lib/src/model.dart]: yes, did it.

**Pull request merged**